client/src/features/Catalogue/components/CatalogFilterPrice/CatalogFilterPrice.js

```diff
@@ -26,25 +24,73 @@ const FilterInput = styled(InputBase)(({ theme }) => ({
   },
 }));
 
+let minTourPrice = 0;
+let maxTourPrice = 0;
+
+function valuetext(value) {
+  return `${value}`;
+}
+
 const CatalogFilterPrice = () => {
-  const [value, setValue] = useState([200, 600]);
+  const dispatch = useDispatch();
+
+  const prices = useSelector((store) => store.filter.prices);
+  const [minPrice, maxPrice] = prices;
+
+  const getAllToursPrices = async () => {
+    try {
+      const { data } = await axios('/products');
+      const allPrices = data.map((tour) => tour.currentPrice);
+      minTourPrice = Math.min.apply(null, allPrices);
+      maxTourPrice = Math.max.apply(null, allPrices);
+      dispatch(setPrices([Math.max(minPrice, minTourPrice), Math.min(maxPrice, maxTourPrice)]));
+      if (maxPrice === 0) {
+        dispatch(setPrices([Math.max(minPrice, minTourPrice), maxTourPrice]));
+      }
+      return allPrices;
+    } catch (err) {
+      return console.log(err);
+    }
+  };
+
+  useEffect(() => {
+    getAllToursPrices();
+  }, []);
```

Comment:
we are using Redux and Redux Thunk.  async functions are written and dispatched in a different way.

we need to talk.....